pallets/identity/src/auth.rs

```diff
@@ -187,7 +194,24 @@ impl<T: Config> Module<T> {
         }
 
         // Run custom per-type validation and updates.
-        accepter(auth.authorization_data, auth.authorized_by)?;
+        let res = accepter(auth.authorization_data.clone(), auth.authorized_by.clone());
+
+        if res.is_err() {
+            // decrement
+            auth.count = auth.count.saturating_sub(1);
+
+            // check if count is zero
+            if auth.count == 0 {
+                <Authorizations<T>>::remove(&target, auth_id);
+                <AuthorizationsGiven<T>>::remove(auth.authorized_by.clone(), auth_id);
```

Comment:
Should emit an event saying that the Auth has been removed due to failed retries.